355 empty meal is being created in admin UI

RLCASTRO opened the pull request:
**Describe the technical changes contained in this PR**
Added a simple form validation for creating new meals in the admin-ui application

**Previous behaviour**
Admin was able to create new meals without providing any inputs for any fields in the form to create a new meal.

**New behaviour**
Admin must provide minimum input to be able to create a new meal. The fields with validation are:

* nameEn


**Related issues addressed by this PR**
Fixes #355

RLCASTRO committed:
added simple HTML validation to the form fields

RLCASTRO committed:
created the validationUtils to validate the form

RLCASTRO committed:
Added validation to the most important fields

RLCASTRO committed:
validationUtils will not be necessary for this solution.

shanthisa commented on admin-ui/src/Meals/MealForm.tsx:
```diff
@@ -1,29 +1,24 @@
-      <TextInput source="nameFr" fullWidth />
+    <SimpleForm >
+      <TextInput source="code" fullWidth validate={required()} />
+      <TextInput source="nameEn" fullWidth validate={required()} />
```
This is correct. We need to make sure the meal plan name in English is a required field. Please check the database and mark only the ones marked as 'not null' to be required. For example: name_Fr is an optional field.

RLCASTRO replied:
Done, left validation for nameEn only, according to the database

shanthisa reviewed (changes requested):
You have taken the right approach. Just need to mark only the required fields as required based on the database

RLCASTRO committed:
Left only the 'not null' fields in the database with form validation

RLCASTRO committed:
removing validation from the categories checkbox

shanthisa commented on .gitignore:
```diff
@@ -3,3 +3,5 @@
 node_modules
 .envrc
+admin-ui/package-lock.json
+mealplanner-ui/src/pages/MealPlans/__generated__/MealPlansQuery.graphql.ts
```
Should not git ignore generated file. Please commit this file.

RLCASTRO replied:
Removed files from the .gitignore. Thank you

shanthisa replied:
I am still seeing that the generated graphql and package-lock.json as part of gitignore

RLCASTRO replied:
sorry, I believe is fixed now. Thank you.

shanthisa reviewed (changes requested):
Please check in generated types. Otherwise any developer has to run 'npm run relay' everytime.

shanthisa commented:
There seems to be conflict in the graphql.ts generated. Can you please rebase v2 and resolve the conflict? Then it will be good to merge.

shanthisa reviewed (dismissed):
LGTM 👍

shanthisa commented:
This has conflicts. Please resolve

shanthisa reviewed (changes requested):
There are 37 files which are all the migrations of the past and favorite meal feature changes which are not part of this ticket. Looks like you have many more files merged into your branch and hence it is wrongly reflecting as your changes. Please take a clean clone of v2 branch and make your changes over it and commit and create a pull request.

RLCASTRO committed:
Revert "Merge branch 'CivicTechFredericton:develop' into 355-Empty-meal-is-being-created-in-admin-ui"

This reverts commit 2c555028efb698a5824764f617d22330eb134aaf, reversing
changes made to 3eed8df9f5d0df9663870dee93c9d6e117ee01ca.

shanthisa committed:
[#543] - Changes related to ingredient and product tables
Added a Route, Link from react-router-dom
The Route routes the endpoints of Ingredients
Custom routes were required to nest ingredients inside Meals
Created IngredientsList page and linked it to specific meal
In IngredientsList page, created links for Create and Edit
Created IngredientCreate and IngredientEdit pages
Hardcoded the meal in ingredients page so as to not allow change of meal name mistakenly
All Ingredients are linked to the specific meal from which they are created
Removed code from meal table
Removed code, cooking_duration, serves from meal list and form (create and edit)
Added cookTime, prepTime, portions in meal list and form - create and edit
Added productKeywords in product form and list pages
Removed code from product form an list pages
Included documentation for MealPlan, SearchMeal, environment

shanthisa commented on admin-ui/src/Meals/MealList.tsx:
```diff
@@ -9,6 +9,7 @@ import {
   ShowButton,
   TextField,
 } from "react-admin";
+import { Link } from "react-router-dom";
```
Why is this required in this PR?

RLCASTRO replied:
solved !

RLCASTRO committed:
removing unecessary code

shanthisa commented on admin-ui/package.json:
```diff
@@ -26,6 +26,6 @@
     "@vitejs/plugin-react": "^1.3.0",
     "typescript": "^4.6.3",
-    "vite": "^2.9.17"
+    "vite": "^2.9.16"
```
Should not be downgraded

RLCASTRO replied:
solved

RLCASTRO committed:
updating the version of the vite in the package manager and the other files related to it after running 'npm i'

shanthisa reviewed (approved):
LGTM 👍
